libproc/lustre.c

```diff
@@ -131,6 +131,42 @@ typedef enum {
  * On EOF, functions return -1 with errno clear.
  */
 
+/*
+ * Fill-in the supplied version components with the Lustre version
+ * found in /sys.
+ *
+ * Returns -1 or less on error, 0 or greater on success.
+ */
+int
+sys_fs_lustre_version (pctx_t ctx, int *major, int *minor, int *patch,
+                       int *fix)
+{
+    int ret = -1;
+    char version[256];
+
+    if ((ret = proc_openf (ctx, PROC_FS_LUSTRE_VERSION)) < 0)
+        goto done;
+
+    if ((ret = proc_gets (ctx, NULL, version, sizeof (version))) < 0)
+        goto close;
+
+    /* first, get the numbers which must be present in a version string */
+    if ((ret = sscanf (version, "%d.%d.%d", major, minor, patch)) != 3) {
```

Comment:
Is it guaranteed that all three values are decimal numbers, and never hex or an arbitrary string?

Reply:
The parsing is equivalent to how we do it in `proc_fs_lustre_version`. As far as I know, the version number string that shows up under `/sys` is the same as it was in `/proc`. If there is a problem with the parsing, I think that's another problem entirely.